### PR TITLE
ffmpeg: handle some API changes for release/4.0 master

### PR DESCRIPTION
**Pull request by aalba6675**
Handle some API changes for newer versions of ffmpeg.

Encountered with Fedora 28 + latest ffmpeg

### CODE REVIEW

File: lib/fix_frame_channel_layout-01.h [hunk 1]
```diff
@@ -5,5 +5,9 @@
 INLINE void fix_frame_channel_layout(AVFrame *frame) {
 	if (frame->channel_layout)
 		return;
+#if LIBAVUTIL_VERSION_MAJOR >= 56
+	frame->channel_layout = av_get_default_channel_layout(frame->channels);
+#else
 	frame->channel_layout = av_get_default_channel_layout(av_frame_get_channels(frame));
+#endif
```
Comment (rfuchs) [inline lib/fix_frame_channel_layout-01.h]: The purpose of the different `fix_frame_channel_layout-*.h` is actually to eliminate the dependency on version test macros. (There seem to be too many different libav/ffmpeg versions out there, with inconsistent version numbers, that I gave up on finding the correct version test for this particular code.) If the `-01.h` fails to compile, it's supposed to move on to `-02.h` and use that if it compiles. Are you sure this is actually needed?

Comment (aalba6675) [reply lib/fix_frame_channel_layout-01.h]: I missed that:  - -01.h actually compiles only the compiler was issuing deprecation warnings.

This version check can be omitted as the code still works.

**Comment by rfuchs** [comment]
See ae3dae7 for an alternative, hopefully handling all cases. Let me know how that works for you.

**Comment by rfuchs** [comment]
Let's see how long it takes before somebody complains that this broke something :)

**Pull request closed without merging**